src/test/run-pass/unsized-locals/by-value-trait-object-safety.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![feature(unsized_locals)]
+
+pub trait Foo {
+    fn foo(self) -> String;
+}
+
+struct A;
+
+impl Foo for A {
+    fn foo(self) -> String {
+        format!("hello")
+    }
+}
+
+
+fn main() {
+    let x = *(Box::new(A) as Box<dyn Foo>);
```

Comment:
Are there preexisting tests for calling such a method directly on a `Box<dyn Foo>` without first dereferencing the box? If not, please add a ui test for that.

Reply:
I've not tested that, and it turned out to compile! I added this as a run-pass test (just below here).

Reply:
awesome!

The RFC isn't very clear on this. But I would see that as a natural consequence.

Can you add more tests (also failure tests ensuring that we can't use the box anymore after having called a method on it?)

Reply:
Thanks!